tests/integration/test_dbfs_store/test_dbfs_store.py

```diff
@@ -0,0 +1,209 @@
+# limitations under the License.
+#
+import os
+import uuid
+from pathlib import Path
+from typing import List
+
+import pandas as pd
+import pytest
+import yaml
+from databricks.sdk import WorkspaceClient
+
+import mlrun
+import mlrun.errors
+from mlrun.utils import logger
+
+PARQUETS_DIR = "/parquets"
+CSV_DIR = "/csv"
+here = Path(__file__).absolute().parent
+config_file_path = here / "test-dbfs-store.yml"
+with config_file_path.open() as fp:
+    config = yaml.safe_load(fp)
+
+test_file_path = here / "test.txt"
+json_path = here / "test_data.json"
+parquet_path = here / "test_data.parquet"
+additional_parquet_path = here / "additional_data.parquet"
+csv_path = here / "test_data.csv"
+additional_csv_path = here / "additional_data.csv"
+with open(test_file_path, "r") as f:
+    test_string = f.read()
+
+MUST_HAVE_VARIABLES = ["DATABRICKS_TOKEN", "DATABRICKS_HOST"]
+
+
+def is_dbfs_configured():
+    env_params = config["env"]
+    for necessary_variable in MUST_HAVE_VARIABLES:
+        if env_params.get(necessary_variable, None) is None:
+            return False
+    return True
+
+
+@pytest.mark.skipif(
+    not is_dbfs_configured(),
+    reason="DBFS storage parameters not configured",
+)
+class TestDBFSStore:
+    def setup_class(self):
+        self._databricks_host = config["env"].get("DATABRICKS_HOST")
+        self.test_root_dir = "/test_mlrun_dbfs_objects"
+        self._object_file = f"file_{str(uuid.uuid4())}.txt"
+        self._object_path = f"{self.test_root_dir}/{self._object_file}"
+        self._dbfs_url = "dbfs://" + self._databricks_host
+        self._object_url = self._dbfs_url + self._object_path
+        self.secrets = {}
+        token = config["env"].get("DATABRICKS_TOKEN", None)
+        self.secrets["DATABRICKS_TOKEN"] = token
+        self.parquets_dir = PARQUETS_DIR
+        self.csv_dir = CSV_DIR
+        self.workspace = WorkspaceClient(host=self._databricks_host, token=token)
+        logger.info(f"Object URL: {self._object_url}")
+
+    @pytest.fixture(autouse=True)
+    def setup_before_each_test(self):
+        all_paths = [file_info.path for file_info in self.workspace.dbfs.list("/")]
+        if self.test_root_dir not in all_paths:
+            self.workspace.dbfs.mkdirs(f"{self.test_root_dir}{self.parquets_dir}")
+            self.workspace.dbfs.mkdirs(f"{self.test_root_dir}{self.csv_dir}")
+        else:
+            self.workspace.dbfs.delete(self.test_root_dir, recursive=True)
+            self.workspace.dbfs.mkdirs(f"{self.test_root_dir}{self.parquets_dir}")
+            self.workspace.dbfs.mkdirs(f"{self.test_root_dir}{self.csv_dir}")
+
+    def teardown_class(self):
+        all_paths_under_test_root = [
+            file_info.path for file_info in self.workspace.dbfs.list(self.test_root_dir)
+        ]
+        for path in all_paths_under_test_root:
+            self.workspace.dbfs.delete(path, recursive=True)
+
+    def _perform_dbfs_tests(self, secrets):
```

Comment:
Need to break this up into multiple tests. You can parameterize them, but I don't think it's actually necessary to run every test once with env and once with secrets.

Reply:
And yeah, I just realized that this pattern is copied from GCS, but still.